prime/client.go

```diff
@@ -0,0 +1,293 @@
+}
+
+func (c *client) GetRooms() (Rooms, error) {
+	response, err := c.sendGetRequest([]string{ComponentRoom})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetRooms()
+}
+
+func (c *client) GetAreas() (Areas, error) {
+	response, err := c.sendGetRequest([]string{ComponentArea})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetAreas()
+}
+
+func (c *client) GetHouse() (*House, error) {
+	response, err := c.sendGetRequest([]string{ComponentHouse})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetHouse()
+}
+
+func (c *client) GetHub() (*Hub, error) {
+	response, err := c.sendGetRequest([]string{ComponentHub})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetHub()
+}
+
+func (c *client) GetShortcuts() (Shortcuts, error) {
+	response, err := c.sendGetRequest([]string{ComponentShortcut})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetShortcuts()
+}
+
+func (c *client) GetModes() (Modes, error) {
+	response, err := c.sendGetRequest([]string{ComponentMode})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetModes()
+}
+
+func (c *client) GetTimers() (Timers, error) {
+	response, err := c.sendGetRequest([]string{ComponentTimer})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetTimers()
+}
+
+func (c *client) GetVinculumServices() (*VinculumServices, error) {
+	response, err := c.sendGetRequest([]string{ComponentService})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetVinculumServices()
+}
+
+func (c *client) GetState() (*State, error) {
+	response, err := c.sendGetRequest([]string{ComponentState})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetState()
+}
+
+func (c *client) GetComponents(components ...string) (*ComponentSet, error) {
+	err := c.validateComponents(components)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := c.sendGetRequest(components)
+	if err != nil {
+		return nil, err
+	}
+
+	return response.GetAll()
+}
+
+func (c *client) GetAll() (*ComponentSet, error) {
+	return c.GetComponents(validComponents...)
+}
+
+func (c *client) RunShortcut(shortcutID int) (*Response, error) {
+	response, err := c.sendSetRequest(ComponentShortcut, shortcutID)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+func (c *client) ChangeMode(mode string) (*Response, error) {
+	response, err := c.sendSetRequest(ComponentMode, mode)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+func (c *client) sendGetRequest(components []string) (*Response, error) {
+	request := &Request{
+		Cmd: CmdGet,
+		Param: &RequestParam{
+			Components: components,
+		},
+	}
+
+	message := fimpgo.NewObjectMessage(CmdPD7Request, "vinculum", request, nil, nil, nil)
+	message.ResponseToTopic = c.responseAddress.Serialize()
+	message.Source = c.clientName
+
+	response, err := c.syncClient.SendReqRespFimp(c.requestAddress.Serialize(), c.responseAddress.Serialize(), message, 5, true)
```

Comment:
I think we should make the timeout configurable. We've seen cases with large sites when 5 seconds are not enough for requesting state.

Reply:
Done.